docs/extending/models.md

```diff
@@ -159,3 +159,20 @@ Dictionary/Map that stores all the additional properties which are not described
 When present, the properties values are store in this backing store instead of using fields for the object. The backing store allows multiple things like dirty tracking of changes, making it possible to get an object from the API, update a property, send that object back with only the changed property and not the full objects. Additionally it will be used for integration with third party data sources.
 
 > **Note:** the backing store is only added if the target language supports it and when the `-b` parameter is passed to the CLI when generating the models.
+
+### Properties and accessors name mangling
+
+To produce a more idiomatic output for specific languages, mangling is applied to the properties names and/or accessors.
+This is the opinionated mangling on each language:
+
+| Language | Property name | Property accessors |
+|---|---|---|
+| CSharp | `PascalCase` | - |
+| Go | - | `PascalCase` |
+| Java | - | `camelCase` |
```

Comment:
Here I would like @EricWittmann feedback as well, the property itself is `private` so, I'm preserving the name and mangling only the accessors, should we mangle also the property name?

Reply:
Not really fussy about this, as those are `private` variables and they do not affect the user API, but I want to explain why I started in this way.
The field I'm "looking at" is named `bootstrap_server_host`, with the encoding in this PR you have this code generated:

```java
private String _bootstrap_server_host;
```
Which is "respecting" the original field but adding a trailing `_`, I find it more easily "searchable"(when comparing to the original spec) than the proposed behavior, which results in:

```java
private String _bootstrapServerHost;
```

I understand that this can be a very personal bias, fortunately, it doesn't affect the public API, so happy to turn this detail in any direction we start to reach an agreement on 🙂

Reply:
I just recompiled the latest `main` and that's what I get ...

To be honest, I'm not really worried about linters, as I believe that they should not run on generated code ... but I understand that this is a biased POV and this project might aim to reduce the noise there.

Reply:
I agree that linters should not be run on generated code. My experience has been that most of the times linting tools and other "advanced" aspects of the project are configured by a senior team member and more junior team members don't have the technical depth to evolve the configuration. And those junior team members often are tasked with new features implementations (adding a new API client would be part of their work).
I think that part of the success path for Kiota is to make sure we ease up adoption and don't generate situations where people's tooling is screaming at them because they only added a new client. Of course we won't be able to factor in all the tools that exist out there, or even custom configurations, but we should try to keep the trivial scenarios in mind.

PS: I think you meant to upload a screenshot in your previous comment.

Reply:
I think @andreaTP is saying that when he recompiled from `main` he was getting generated code with private fields containing a leading `_`.  

I love the goal of generating code that would trigger as few warnings from linters as possible!  And while I agree that linters should ideally not be run on generated code, it happens all the time.  Right now in Apicurio Registry we have some protobuf to Java and Avro to Java codegen going on in a few places (mostly tests I think) - and my IDE **desperately** wants to tell me about all of the unused variables and imports and deprecated APIs the generated code is using.  It's very annoying.

In this case I think the right output from the generator for a field should be:

```java
private String bootstrapServerHost;
```

Standard java `camelCase` output with no leading `_`.  But understand I don't have nearly as much context as I might need for this opinion.  :)

Reply:
Here you go:
![Screenshot 2023-01-31 at 19 00 13](https://user-images.githubusercontent.com/5792097/215857070-47427c09-1a73-433a-a565-bfde68531a60.png)
![Screenshot 2023-01-31 at 19 00 28](https://user-images.githubusercontent.com/5792097/215857073-4b1ce9fb-5cbb-4023-a63f-1fc8e3a7a5f2.png)

according to the indication this should be the expected outcome.